### PR TITLE
#226 extensive documentation

### PR DESCRIPTION
**Pull request by Dschoni**
This pull request is meant to introduce numpydoc-style documentation into the project.

### CODE REVIEW

File: vertica_python/vertica/connection.py [hunk 1]
```diff
@@ -68,7 +68,81 @@
...
+        The name of the database to connect to. Default is the same as for `user`.
+    - host : string
+        The ip adress or hostname of the database. Default is 'localhost'.
+    - log_level : int
+        The severity level of messages to be logged. Available logging levels
+        and their corresponding integer values are:
+            =============  =======
+            Logging Level  Integer
+            =============  =======
+            CRITICAL        50
```
Comment (darwinyip) [inline vertica_python/vertica/connection.py]: Align with the other values

File: vertica_python/vertica/connection.py [hunk 2]
```diff
@@ -166,7 +240,77 @@ def peek(self):
...
 class Connection(object):
+    """The connection class handles the establishment of connections to the database
+    and offers methods to handle connection-specific tasks such as commits, rollbacks etc.
+    """
     def __init__(self, options=None):
+        """
+        A connection object is being created by passing a `options` dictionary
+        into this method. This dictionary holds the parameters as defined below.
+
+        Parameters
```
Comment (darwinyip) [inline vertica_python/vertica/connection.py]: Align with block below

File: vertica_python/vertica/connection.py [hunk 3]
```diff
@@ -489,16 +649,16 @@ def process_message(self, message):
     def __str__(self):
         safe_options = {key: value for key, value in self.options.items() if key != 'password'}
 
-        s1 = "<Vertica.Connection:{0} parameters={1} backend_pid={2}, ".format(
```
Comment (darwinyip) [inline vertica_python/vertica/connection.py]: seems unnecessary

File: vertica_python/vertica/connection.py [hunk 4]
```diff
@@ -166,7 +240,77 @@ def peek(self):
...
+        The name of the database to connect to. Default is the same as for `user`.
+    - host : string
+        The ip adress or hostname of the database. Default is 'localhost'.
+    - log_level : int
+        The severity level of messages to be logged. Available logging levels
+        and their corresponding integer values are:
+            =============  =======
+            Logging Level  Integer
+            =============  =======
+            CRITICAL        50
```
Comment (darwinyip) [inline vertica_python/vertica/connection.py]: Align with rest of numbers

**Comment by sitingren** [comment]
This PR is incomplete and inactive for a long time, close for now. Please open a new PR when you are ready. Thanks!

**Pull request closed without merging**